orchestrator/cli/generator/generator/enums.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019-2020 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def get_non_standard_fields(fields: list[dict]) -> str:
+    return ", ".join(type for field in fields if (type := field["type"]) not in ["enum", "int", "bool", "str"])
+
+
+def is_enum(field: dict) -> bool:
+    return field["type"] == "enum"
+
+
+def is_enum_of_type(field: dict, enum_type: str) -> bool:
+    return is_enum(field) and field["enum_type"] == enum_type
+
+
+def is_str_enum(field: dict) -> bool:
+    return is_enum_of_type(field, "str")
+
+
+def is_int_enum(field: dict) -> bool:
+    return is_enum_of_type(field, "int")
+
+
+def has_default(field: dict) -> bool:
+    return "default" in field
+
+
+def convert_enum(field: dict, separator: str = ".") -> dict:
+    return (
```

Comment:
Sometimes I use an inner generator function to make this a bit more readable. You end up with something like:

```python
def convert_enum(field: dict, separator: str = ".") -> dict:
  def _convert_enum() -> Generator:
     yield from field.items
     enum_type = field["name"].title().replace("_", ""))
     yield "type", enum_type
     if has_default_field(field)
        yield "default", enum_type + separator + str(field["default"])

    return dict(_convert_enum())
```

Reply:
I see your point, but I'm not sure if this will increase readability 😃